app/services/save_provider_user.rb

```diff
@@ -13,15 +13,21 @@ def call!
 private
 
   def update_permissions
+    unselected_permissions = ProviderPermissions
+      .where(provider_user: @provider_user)
+      .where.not(provider_id: @permissions.values.flatten)
+
     ActiveRecord::Base.transaction do
-      ProviderPermissions
-        .where(provider_user: @provider_user)
-        .update_all(ProviderPermissionsOptions.reset_attributes)
+      ProviderPermissionsOptions::VALID_PERMISSIONS.each do |permission|
+        unselected_permissions.where(permission => true).each { |perm| perm.update(permission => false) }
```

Comment:
To avoid silent failures if validation fails:

```suggestion
        unselected_permissions.where(permission => true).each { |perm| perm.update!(permission => false) }
```